crates/iota/src/iota_commands.rs

```diff
@@ -505,18 +505,22 @@ impl IotaCommand {
             } => {
                 match &mut cmd {
                     iota_move::Command::Build(build) if build.dump_bytecode_as_base64 => {
-                        // `iota move build` does not ordinarily require a network connection.
-                        // The exception is when --dump-bytecode-as-base64 is specified: In this
-                        // case, we should resolve the correct addresses for the respective chain
-                        // (e.g., testnet, mainnet) from the Move.lock under automated address
-                        // management.
-                        let config =
-                            client_config.unwrap_or(iota_config_dir()?.join(IOTA_CLIENT_CONFIG));
-                        prompt_if_no_config(&config, false, true).await?;
-                        let context = WalletContext::new(&config, None, None)?;
-                        let client = context.get_client().await?;
-                        let chain_id = client.read_api().get_chain_identifier().await.ok();
-                        build.chain_id = chain_id.clone();
+                        if build.ignore_chain {
+                            build.chain_id = None;
```

Comment:
Could move this assignment up if you want

Reply:
I think it's clearer this way

Reply:
I'm also wondering if the chain_id is set to `None` to begin with. If so, this whole branch is not needed

Reply:
It's an Option that can be set to Some by the caller

Reply:
So if the callers sets it and also sets `--ignore-chain` then we delete it? Seems a tiny bit weird

Reply:
Ah it can also read from a config, this is the original PR description:
```
Allow sui move build --dump-bytecode-as-base64 to not need to read the chain from client.yaml by adding a flag --ignore-chain. This allows building to proceed without a network connection or active environment, but it will not be able to automatically determine the addresses of its dependencies.
NB: --ignore-chain depends on dump-bytecode-as-base64, so it cannot be used on its own.
```

Reply:
Okay, thanks for the explanation